### ISSUE TITLE
General optimization: World geometry and skybox brushes

### ISSUE THREAD

**Issue by jethack** [dasllama/de_office#9]
Before this issue is solved, it's impossible to take on issues #5 or #4, because they completely rely on world geometry.

Currently it appears that almost the entire level is built with **func_detail**, which does not block visibility. This means that in its current state, no BSP optimization is applied to the map *at all*. In order for any optimization to happen, there needs to be proper, simplified world geometry that blocks visibility. 

Because of how the Source engine works, you should build levels almost as if they were Doom (1994) corridors - the map needs to have as little unnecessary "inside the sky" space as possible. If you can't see a part of the map and you don't need to throw grenades through it, it should not be "inside" the level but "outside", touching "the void".

The sky brushes around the map are currently a box, which is also not the proper way to implement a  skybox. It limits optimization because it creates those unnecessary spaces.

For further reading/watching: 
https://developer.valvesoftware.com/wiki/Visibility_optimization
https://www.youtube.com/watch?v=mIy0Usc8zXU
https://www.youtube.com/watch?v=swBfUW-Bsxs

**Comment by dasllama** [comment]
@jethack Yes. And some mappers say that I should use func_detail so I build up the map with that. Maybe I can optimize the geometry by adding nodraw block to the level, where the void could be, so that we can optimize the skybox.

**Comment by jethack** [comment]
@dasllama Yes, this is totally possible. It's just a lot of work redoing all the geometry

**Comment by dasllama** [comment]
@jethack so I'll do that.
